File: src/rbry/corelib-aot/str.js
;
;
var str = (() =>{
        function str() {
            return Object.create(str.prototype);
        }
    str.prototype.__new__ = function __new__(o, ){const cls = this;
        return $RB.call(__JS_AS_STRING__, o, );

    };
str.prototype.__eq__ = function __eq__(o, ){const self = this;
        return self == o;

    };
str.prototype.__mul__ = function __mul__(o, ){const self = this;
        {const tname = type(o).name;if( tname === "int") {
            return self.repeat(Number(b));

        }else {
            return NotImplemented;

    }};

    };
str.prototype.__rmul__ = function __rmul__(o, ){const self = this;
        {const tname = type(o).name;if( tname === "int") {
            return self.repeat(Number(b));

        }else {
            return NotImplemented;

    }};

    };

        return str;
    })();
    ;

const __exported__ = {str}

File: src/rbry/corelib/primitives/str.py
from types import NotImplementedType
from RBM import __JS_OP__, __JS_SHADOW__, string

@__JS_SHADOW__("string")
class str(string):
    def __new__(cls, o: object, /) -> str:
        #return __JS_AS_STRING__(o) # type: ignore
        return type(o).__str__(o) # type: ignore
    
    def __eq__(self, o: object, /) -> bool:
        return __JS_OP__(self, "==", o)
    
    def __add__(self, o: object, /) -> NotImplementedType|str:
        match o:
            case str  (): return __JS_OP__(self, "+", o) # type: ignore
            case _      : return NotImplemented

    def __mul__(self, o: object, /) -> NotImplementedType|str:
        match o:
            case int  (): return __JS_WRITE__("self.repeat(Number(b))") # type: ignore
            case _      : return NotImplemented
    
    def __rmul__(self, o: object, /) -> NotImplementedType|str:
        match o:
            case int  (): return __JS_WRITE__("self.repeat(Number(b))") # type: ignore
            case _      : return NotImplemented